src/snappy_device_agents/devices/maas2/maas2.py

```diff
@@ -72,7 +75,12 @@ def provision(self):
         distro = provision_data.get("distro", "xenial")
         kernel = provision_data.get("kernel")
         user_data = provision_data.get("user_data")
-        self.deploy_node(distro, kernel, user_data)
+        storage_data = provision_data.get("disks")
+        if storage_data:
```

Comment:
if the disks section does not exist, then we will never create maas_storage and the call to deploy_node() below will get an error.  Please make sure to test this with *and* without storage specified.

Reply:
Is there a piece of this that also needs to always ensure we reset the disk back to flat? regardless of there being storage data or not

i also prefer to be explicit "storage_data is not None"

Reply:
I did it this way originally, and thought it'd be clever to pass the constructor instead of storage_data. It makes more sense to revert back to defining the constructor in init and passing storage_data in the call to deploy node.